src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.cs

```diff
@@ -100,5 +106,28 @@ public static partial class JsonSerializer
 
             return;
         }
+
+        private static ReadOnlySpan<byte> GetUnescapedString(ReadOnlySpan<byte> utf8Source, int idx)
```

Comment:
This doesn't return a string, rename.

Also, does `reader.GetString()` not work here because of the string allocation?

Looking in `JsonClassInfo.GetProperty`, you effectively do that again anyway:
https://github.com/dotnet/corefx/blob/be77af0a58a0ea8399f93d86d3338f10095df104/src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs#L142-L147

Reply:
As an aside, this helper/code duplication won't be necessary if we have an API like this, within the reader: https://github.com/dotnet/corefx/issues/35386